lib/dotcom/realtime_schedule.ex

```diff
@@ -25,10 +24,12 @@ defmodule Dotcom.RealtimeSchedule do
 
   @predicted_schedules_per_stop 2
 
+  @predictions_repo Application.compile_env!(:dotcom, :repo_modules)[:predictions]
+
   @default_opts [
     stops_fn: &StopsRepo.get/1,
     routes_fn: &RoutesRepo.by_stop_with_route_pattern/1,
-    predictions_fn: &PredictionsRepo.all_no_cache/1,
+    predictions_fn: Function.capture(@predictions_repo, :all_no_cache, 1),
```

Comment:
We should just remove the dependency injection. It's pretty simple to just replace all instances of `predictions_fn.(params)` with a call to `@predictions_repo.all_no_cache(params)`.